File: duplicate_letter_string/duplicate_letter.py
import re
def duplicate_letter(string, object = {}):
  new_object = object
  new_string = re.sub("\s", "", string)
  if (new_string != ""):
    if (new_string[0] in new_object):
      new_object[new_string[0]] += 1
      return duplicate_letter(new_string[1:len(new_string)], new_object)
    else:
      new_object[new_string[0]] = 1
      return duplicate_letter(new_string[1:len(new_string)], new_object)
  else:
    return new_object

print(duplicate_letter("ethan michael gustafson python", {}))

File: duplicate_letter_string/duplicate_letter.rb
def duplicate_letter(string, obj = {})
  new_string = string.gsub(/\s/, "") # this removes whitespace between words

  if new_string != ""
    if !!obj.has_key?(string[0].to_sym)
      obj[new_string[0].to_sym] += 1
      duplicate_letter(new_string.slice(1, string.length), obj)
    else
      obj[new_string[0].to_sym] = 1
      duplicate_letter(new_string.slice(1, string.length), obj)
    end
  else
    obj
  end
  obj
end

pp duplicate_letter("ethan michael gustafson ruby")